mlrun/model_monitoring/controller.py

```diff
@@ -71,14 +71,23 @@ def _get_last_analyzed(self) -> Optional[int]:
         except HttpResponseError as err:
             logger.warn(
                 "Failed to get the last analyzed time for this endpoint and application, "
-                "as this is probably the first time this application is running. "
-                "Using the first request time instead.",
+                "as this is probably the first time this application is running. ",
+                "Using the latest between first request time or last update time instead.",
                 endpoint=self._endpoint,
                 application=self._application,
                 first_request=self._first_request,
                 error=err,
             )
-            return self._first_request
+
+            return max(
+                self._first_request,
+                int(
+                    (
+                        datetime.datetime.fromtimestamp(self._stop)
+                        - datetime.timedelta(days=1)
+                    ).timestamp()
+                ),
```

Comment:
To avoid timezones mess from `fromtimestamp`:
```suggestion
                self._stop - int(datetime.timedelta(days=1).total_seconds()),
```
https://docs.astral.sh/ruff/rules/call-datetime-fromtimestamp/